### PR TITLE
Add pa_mt_lebanon spider

### PR DESCRIPTION
**Pull request by maxachis**

### CODE REVIEW

**Comment by ben-nathanson** [comment]
For consistency, this spider should be renamed to `mt_lebo_public_meetings`. `pa_mt_lebo_public_meetings` lumps it in with other state-wide agencies and could potentially be confusing.

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        return COMMISSION
+
+    def _parse_start(self, response, index):
+        monthAndDay = response.xpath("//tbody/tr/td/text()").getall()[index]
+        day = self.getDay(monthAndDay)
+        month = self.getMonth(monthAndDay)
+        monthAndDay = self.getMonthAndDay(monthAndDay)
+        year = str(self.getYear(response))
+        time = self.getTime(response)
+        """Parse start datetime as a naive datetime object."""
```
Comment (ben-nathanson) [inline city_scrapers/spiders/pa_mt_lebo_public_meetings.py:70]: For consistency move those comment up to line 64 / the first line after the method definition.

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        """Seems like the meeting is always in the same place given the info in the Agenda PDFs."""
+        return {
+            "address": ADDRESS,
+            "name": LOCATION_NAME,
+        }
+
+    def _parse_links(self, response, index):
+        href = "http://mtlebanon.org/299/Commission-Meetings"
+        title = "Mt Lebanon Commission Meeting"
+        """Parse or generate links."""
```
Comment (ben-nathanson) [inline city_scrapers/spiders/pa_mt_lebo_public_meetings.py:98]: Same comment here - move this comment up

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        return [{"href": href, "title": title}]
+
+    def _parse_source(self, response, index):
+        """Parse or generate source."""
+        return response.url
+
+    def isValidEvent(self, response, index):
+        # Check if cell contains Date and Time
+        pattern = "[Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec]\S* \d+"
```
Comment (ben-nathanson) [inline city_scrapers/spiders/pa_mt_lebo_public_meetings.py:107]: Make this into a constant

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        eventString = response.xpath("//tbody/tr/td/text()").getall()[index]
+        if re.search(pattern, eventString) != None:
+            # Check if cell contains asterisks, which indicate special events
+            #  which for now will be ignored
+            pattern = "\*\*"
```
Comment (ben-nathanson) [inline city_scrapers/spiders/pa_mt_lebo_public_meetings.py:112]: Make this into a constant

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            if re.search(pattern, eventString) != None:
```
Comment (ben-nathanson) [inline city_scrapers/spiders/pa_mt_lebo_public_meetings.py:113]: For readability, make this into an if-else.

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+    def getTime(self, response):
+        """Retrieves time in AM/PM format. As of this writing, it is 8 p.m."""
+        timepath = '//*[@id="divEditor04c7e1eb-6543-4734-b9bb-f65cd6b048c3"]/ul[2]/li[1]/text()'
+        time = response.xpath(timepath).get(0)
+        groups = re.search("(\d+ [a-z])\.([a-z])\.", time).groups()
+        return groups[0].upper() + groups[1].upper()
+
+    def getMonth(self, monthAndDay):
+        pattern = "[Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec]\S*"
```
Comment (ben-nathanson) [inline city_scrapers/spiders/pa_mt_lebo_public_meetings.py:126]: Make this into a constant

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+        return re.search(pattern, monthAndDay).group(0)
+
+    def getMonthAndDay(self, monthAndDay):
+        pattern = "[Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec]\S* (\d+)"
```
Comment (ben-nathanson) [inline city_scrapers/spiders/pa_mt_lebo_public_meetings.py:130]: Make this into a constant

File: city_scrapers/spiders/pa_mt_lebo_public_meetings.py (hunk 6 above)
Comment (ben-nathanson) [reply city_scrapers/spiders/pa_mt_lebo_public_meetings.py:126]: There's an opportunity for code reuse here

**Review by ben-nathanson — CHANGES_REQUESTED** [review]
When I ran `scrapy crawl pa_mt_lebanon` I found an `IndexError` click [here](https://pastebin.com/hhzsNyvH) for logs.

**Comment by ben-nathanson** [comment]
There are 46 files associated with this - mostly related to tests - can we get rid of any them without affecting regression tests? If I takes more than 20 minutes don't worry about it but that's a lot of info.

**Comment by maxachis** [comment]
> There are 46 files associated with this - mostly related to tests - can we get rid of any them without affecting regression tests? If I takes more than 20 minutes don't worry about it but that's a lot of info.

After looking into this, the answer is yes, with the caveat that unless we modify the HTML file itself, we can't remove the  "pa_mt_lebanon_files" directory. The HTML file references that _a lot_--46 times--and even if it doesn't get the files in the folder, so long as it can reference the folder itself, it functions fine. If the folder isn't there, though, it throws an error. 

If it seems stupid to keep an empty folder, the answer is yes, but it's either that or painstakingly removing the HTML references, which I'm willing to do if necessary (especially since it might confuse people working on this later on as to why we have an empty folder), but which seems so tightly wound in the thoroughly convoluted HTML code that resolving that might be more trouble than it's worth.

**Review by ben-nathanson — CHANGES_REQUESTED** [review]
This looks good! Just be sure to get the naming consistent with the other spiders:
- rename the class and variables
- rename the file/directory names

If it would be simpler, you could just do `scrapy genspider ...` and copy the code into the newly named files/directories/variables.

File: city_scrapers/spiders/mt_lebo_public_meetings.py [hunk 8]
```diff
@@ -14,6 +14,9 @@
 ADDRESS = ("710 Washington Road, Pittsburgh, PA 152289",)
 LOCATION_NAME = ("Municipal Building",)
 
+monthAndDayPattern = "[Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec]\S* \d+"
```
[the remark this answers is not in the document]
Comment (maxachis) [reply city_scrapers/spiders/mt_lebo_public_meetings.py:17]: Thank you! I'm quickly growing quite attached to Regex.

**Comment by maxachis** [comment]
> When I ran `scrapy crawl pa_mt_lebanon` I found an `IndexError` click [here](https://pastebin.com/hhzsNyvH) for logs.

This was related to what I was saying about the rickety nature of the website -- and hence, at least previously, the spider. I modified the spider to be a bit more flexible. In this case, it's looking more for keywords rather than specific id's, which are either sparse or a sequence of near-random-seeming characters.

**Review by ben-nathanson — APPROVED** [review]
Looks good! Tested locally. Thank you for incorporating my suggestions about naming conventions. IMO they can be a bit pedantic but it helps keep a clean house. 

🚢 it.

**Comment by bonfirefan** [comment]
Max, great spider. I would note that you can check for month date patterns without having to specify them - you could use something clever with datetime's `%b` tag which recognize abbreviated months like Jan / Feb / Mar

**Pull request merged**